compaction/time_window_compaction_strategy.cc

```diff
@@ -221,22 +221,24 @@ time_window_compaction_strategy::get_sstables_for_compaction(table_state& table_
         return compaction_descriptor();
     }
 
-    // Find fully expired SSTables. Those will be included no matter what.
-    std::unordered_set<shared_sstable> expired;
-
     if (db_clock::now() - _last_expired_check > _options.expired_sstable_check_frequency) {
         clogger.debug("[{}] TWCS expired check sufficiently far in the past, checking for fully expired SSTables", fmt::ptr(this));
+
+        // Find fully expired SSTables. Those will be included no matter what.
+        std::unordered_set<shared_sstable> expired;
         expired = table_s.fully_expired_sstables(candidates, compaction_time);
```

Comment:
Nit: might as well change to `auto expired = ...`

Reply:
This is done in a following patch.
I implemented it this way in this patch just to minimize the diff and make the logic changes clearer.